### PR TITLE
:seedling: Create framework images for each flavor

### PR DESCRIPTION
**Pull request by mudler**
This is an alternative to #972 and supersedes it.

Framework images now contains the whole framework:
- kairos-agent
- elemental-cli
- luet
- base cloud configs
- grub configuration files
- services for inits

The images are pushed during release as:

```
quay.io/kairos/framework:<version>_<flavor>
```

So can be used directly in Dockerfiles as such:

```Dockerfile
FROM fedora:36

## Copy the framework image depending on the distro family
COPY --from=quay.io/kairos/framework:v2.0.0_fedora / /

## Install kernel and packages..
RUN ...

## Generate initrd
RUN kernel=$(ls /boot/vmlinuz-* | head -n1) && \
            ln -sf "${kernel#/boot/}" /boot/vmlinuz
RUN kernel=$(ls /lib/modules | head -n1) && \
            dracut -v -N -f "/boot/initrd-${kernel}" "${kernel}" && \
            ln -sf "initrd-${kernel}" /boot/initrd && depmod -a "${kernel}"
RUN rm -rf /boot/initramfs-*
```

Part of https://github.com/kairos-io/kairos/issues/755

### CODE REVIEW

File: .github/workflows/image-arm.yaml [hunk 1]
```diff
@@ -11,20 +11,33 @@ concurrency:
...
       fail-fast: false
-      matrix:
-        include:
-          - flavor: "opensuse-leap-arm-rpi"
-            model: rpi64
-          - flavor: "opensuse-tumbleweed-arm-rpi"
-            model: rpi64
-          - flavor: "alpine-arm-rpi"
-            model: rpi64
+      matrix: ${{fromJson(needs.get-matrix.outputs.matrix)}}
```
Comment (mauromorales) [inline .github/workflows/image-arm.yaml:40]: very cool!

File: overlay/files/system/oem/06_recovery.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+         group: 0
+   boot:
+     - name: "Recovery"
+       if: '[ -f "/run/cos/recovery_mode" ]'
+       hostname: "cos-recovery"
+       commands:
+       - |
+            source /etc/os-release
+            echo >> /etc/issue
+            echo "You are booting from recovery mode. Run 'cos-reset' to reset the system to $VERSION" >> /etc/issue
```
Comment (mauromorales) [inline overlay/files/system/oem/06_recovery.yaml:28]: does this version need to be evaluated? or is just as an example?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -32,40 +52,26 @@ jobs:
           IMAGE: quay.io/kairos/framework
         run: |
           export TAG=${GITHUB_REF##*/}
-          ./earthly.sh +framework-image --IMG=${IMAGE}:$TAG --FLAVOR=${FLAVOR} --WITH_KERNEL=false
-          ./earthly.sh +framework-image --IMG=${IMAGE}-generic:$TAG --FLAVOR=${FLAVOR} --WITH_KERNEL=true
+          earthly --ci --push +build-framework-image --FLAVOR=${FLAVOR}
```
Comment (mauromorales) [inline .github/workflows/release.yaml:55]: should this push every single time or only on master?

File: overlay/files/system/oem/06_recovery.yaml (hunk 2 above)
Comment (mudler) [reply overlay/files/system/oem/06_recovery.yaml:28]: this is not really part of this change, as I have just copied the files from https://github.com/kairos-io/packages/tree/main/packages/system/cloud-config/oem here, however it needs to be evaluated (that's why it sources /os-release)

File: .github/workflows/release.yaml (hunk 3 above)
Comment (mudler) [reply .github/workflows/release.yaml:55]: ok should be good now, the `release.yaml` pushes always (as it is triggered on tags from branches), while the `image.yaml` pushes only on master

**Review by mauromorales — DISMISSED** [review]
lgtm

**Comment by mudler** [comment]
somehow, alpine is broken, having a look at it..

**Comment by mudler** [comment]
Found the reason I think, fixed in https://github.com/kairos-io/kairos/pull/973/commits/00074e0c300b7eda7c8ab5f4fb8aca213e24464c. Should be good to go

**Comment by mudler** [comment]
@mauromorales need another review here :pray:

File: .github/workflows/release-arm.yaml [hunk 4]
```diff
@@ -1,27 +1,42 @@
...
+      - uses: actions/checkout@v3
+      - run: |
+          git fetch --prune --unshallow
+      - id: set-matrix
+        run: |
+          content=`cat ./.github/flavors-arm.json`
+          # the following lines are only required for multi line json
+          # the following lines are only required for multi line json
+          # the following lines are only required for multi line json
+          # the following lines are only required for multi line json
```
Comment (mauromorales) [inline .github/workflows/release-arm.yaml:21]: does it need each of these? or just one?

Comment (mauromorales) [reply .github/workflows/release-arm.yaml:21]: saw from the other file it was only once so I assume only one is needed

File: .github/workflows/release-arm.yaml [hunk 5]
```diff
@@ -1,27 +1,42 @@
...
+          content="${content//'%'/'%25'}"
+          content="${content//$'\n'/'%0A'}"
+          content="${content//$'\r'/'%0D'}"
+          # end of optional handling for multi line json
```
Comment (mauromorales) [inline .github/workflows/release-arm.yaml:25]: same here

File: overlay/files/system/oem/07_live.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# Before you change this file manually,
+# consider copying this file to /usr/local/cloud-config or
```
Comment (mauromorales) [inline overlay/files/system/oem/07_live.yaml:2]: I think this sentence got cut

```suggestion
# consider copying this file to /usr/local/cloud-config or
# copy the file with a prefix starting by 90, e.g. /oem/91_custom.yaml
```

Comment (mauromorales) [reply overlay/files/system/oem/07_live.yaml:2]: ah it's also bellow, removing my comment

**Review by mauromorales — APPROVED** [review]
some tiny duplication in the comments but should not block from merging

File: .github/workflows/release-arm.yaml (hunk 4 above)
Comment (mudler) [reply .github/workflows/release-arm.yaml:21]: damn it, is the YAML formatter.. it injected those somehow :(

Comment (mudler) [reply .github/workflows/release-arm.yaml:21]: if this turns out to be green I'll merge and amend in another PR, guessing maybe there are other pieces to fix too..

**Pull request merged**